File: backend/voting-app/src/main/java/com/tu/votingapp/dto/response/elections/VoteResponseDTO.java
package com.tu.votingapp.dto.response.elections;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class VoteResponseDTO {
    private Long id;
    private Long userId;
    private Long electionId;
    private Long candidateId;    // nullable if a party vote
    private Long partyId;        // nullable if a candidate vote
    private LocalDateTime voteTimestamp;
}

File: frontend/android/app/src/main/java/com/example/android/dto/response/elections/VoteResponseDTO.kt
package com.example.android.dto.response.elections

import kotlinx.serialization.Serializable

@Serializable
data class VoteResponseDTO(
    val id: Long,
    val userId: Long,
    val electionId: Long,
    val candidateId: Long?,
    val partyId: Long?,
    val voteTimestamp: String
)